packages/edit-site/src/components/editor/index.js

```diff
@@ -50,6 +50,41 @@ const interfaceLabels = {
 	footer: __( 'Editor footer' ),
 };
 
+function useIsSiteEditorLoading() {
+	const { isLoaded: hasLoadedPost } = useEditedEntityRecord();
+	const { hasResolvingSelectors } = useSelect( ( select ) => {
+		return {
+			hasResolvingSelectors: select( coreStore ).hasResolvingSelectors(),
+		};
+	} );
+	const [ loaded, setLoaded ] = useState( false );
+	const timeoutRef = useRef( null );
+
+	useEffect( () => {
+		if ( ! hasResolvingSelectors && ! loaded ) {
+			clearTimeout( timeoutRef.current );
```

Comment:
This `clearTimeout` looks redundant. And the `timeoutRef` could be a local variable inside the effect callback.

This `clearTimeout` can be executed only when:
- the effect mounts initially, and in that case there is no timeout to clear yet
- a dependency, either `hasResolvingSelectors` or `loaded`, has changed, and in that case the timeout was cleared by the cleanup function of the previous effect.

Reply:
That makes sense, thanks! Addressed in 23c51d6e4229b53d5d2149a69c78fa4ea6954439 and b92c69b13f4f05ed762370fed2c17ed276451a08.